python/ray/ml/config.py

```diff
@@ -105,6 +115,50 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
+        allowed_keys: Iterable[str],
+        scaling_config_arg_name: str,
+        caller_name: str,
```

Comment:
Can we get rid of this? We shouldn't add arguments just for a print output. Instead, the enclosing function should raise the respective error. That function can just state the allowed keys. Using `raise from` will preserve information about which key is affected in the stacktrace.

Reply:
will be done in https://github.com/ray-project/ray/pull/23889